platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventSerializationUtils.java

```diff
@@ -0,0 +1,296 @@
+     */
+    @NonNull
+    private static UnsignedEvent deserializeUnsignedEvent(@NonNull final SerializableDataInputStream in)
+            throws IOException {
+        Objects.requireNonNull(in, "The input stream must not be null");
+        final int version = in.readInt();
+        if (version != UNSIGNED_EVENT_VERSION) {
+            throw new IOException("Unsupported version: " + version);
+        }
+
+        final SoftwareVersion softwareVersion =
+                in.readSerializable(StaticSoftwareVersion.getSoftwareVersionClassIdSet());
+
+        final NodeId creatorId = in.readSerializable(false, NodeId::new);
+        if (creatorId == null) {
+            throw new IOException("creatorId is null");
+        }
+        final EventDescriptorWrapper selfParent = EventDescriptorWrapper.deserialize(in);
+        final List<EventDescriptorWrapper> otherParents = EventDescriptorWrapper.deserializeList(in);
+        final long birthRound = in.readLong();
+
+        final Instant timeCreated = in.readInstant();
+        in.readInt(); // read serialized length
+        final List<EventTransaction> transactionList = new ArrayList<>();
+        final int transactionSize = in.readInt();
+        if (transactionSize > 0) {
+            in.readBoolean(); // allSameClass
+        }
+        for (int i = 0; i < transactionSize; i++) {
+            final long classId = in.readLong();
+            final int classVersion = in.readInt();
+            if (classId == APPLICATION_TRANSACTION_CLASS_ID) {
+                final OneOf<TransactionOneOfType> oneOf = new OneOf<>(
+                        TransactionOneOfType.APPLICATION_TRANSACTION,
+                        deserializeApplicationTransaction(in, classVersion));
+                transactionList.add(new EventTransaction(oneOf));
+            } else if (classId == STATE_SIGNATURE_CLASS_ID) {
+                final OneOf<TransactionOneOfType> oneOf = new OneOf<>(
+                        TransactionOneOfType.STATE_SIGNATURE_TRANSACTION,
+                        deserializeStateSignatureTransaction(in, classVersion));
+                transactionList.add(new EventTransaction(oneOf));
+            } else {
+                throw new IOException("Unknown classId: " + classId);
+            }
+        }
+
+        return new UnsignedEvent(
+                softwareVersion, creatorId, selfParent, otherParents, birthRound, timeCreated, transactionList);
+    }
+
+    @Nullable
+    private static Bytes deserializeApplicationTransaction(
+            @NonNull final SerializableDataInputStream in, final int classVersion) throws IOException {
+        if (classVersion != APPLICATION_TRANSACTION_VERSION) {
+            throw new IOException("Unsupported application class version: " + classVersion);
+        }
+        final byte[] bytes = in.readByteArray(MAX_ARRAY_LENGTH);
+
+        if (bytes != null) {
+            return Bytes.wrap(bytes);
+        }
+        return null;
```

Comment:
Does this exit point need a unit test?  Code coverage warning says it is untested.  Are there real world conditions where null can be returned here?

Reply:
nope, change it so it never returns null